### PR TITLE
[BPK-1261] Add eslint config to backpack-ios

### PR DESCRIPTION
**Pull request by georgegillams**
Having some issues getting `lint-staged` to run `eslint` when committing a modified `.js` file. Apart from that, this should be ready to ship!

### CODE REVIEW

**Comment by matthewdavidson** [comment]
FYI prettier wasn't even in scope for this - just eslint :)

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+          ".json"
+        ]
+      }
+    }
+  },
+  "rules": {
+    "prettier/prettier": "error",
+    "no-console": "off",
+    "strict": ["error", "global"],
+    "curly": "warn"
```
Comment (matthewdavidson) [inline .eslintrc]:
```
"strict": ["error", "global"],
"curly": "warn"
```

These should be removed

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  "parser": "babel-eslint",
+  "extends": [
+      "skyscanner",
+      "prettier"
+  ],
+  "plugins": [
+    "prettier"
+  ],
+  "env": {
+    "browser": true,
```
Comment (matthewdavidson) [inline .eslintrc]: Non of the js code in this repo is for the browser. Im pretty sure we dont have any jest tests either.

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
+  "parser": "babel-eslint",
```
Comment (matthewdavidson) [inline .eslintrc]: the parser is already declared in `eslint-config-skyscanner`

File: .eslintrc [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  ],
+  "plugins": [
+    "prettier"
+  ],
+  "env": {
+    "browser": true,
+    "jest": true
+  },
+  "settings": {
+    "import/resolver": {
```
Comment (matthewdavidson) [inline .eslintrc]: Is this necessary?

**Comment by shaundon** [comment]
@matthewdavidson thanks for the comments - I've updated it now :)

**Pull request merged**